### PR TITLE
(DOCSP-9912): MongoDB Realm Backend Tutorial

### PR DESCRIPTION
**Pull request by mongomoe**
jira
---
https://jira.mongodb.org/browse/DOCSP-9912

stage
-----
https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-9912-Realm-Backend-Tutorial/tutorial/backend.html

### CODE REVIEW

**Comment by cbush** [comment]
Another thing: "What repo?" I think you should explicitly mention that the user should create a repo just for this.

**Comment by cbush** [comment]
Sorry, one more thing. I get a deprecation notice email from GitHub saying:

>Please use the Authorization HTTP header instead, as using the `access_token` query parameter is deprecated. If this token is being used by an app you don't have control over, be aware that it may stop working as a result of this deprecation.

Any way we can just do this in the tutorial?

**Comment by mongomoe** [comment]
@cbush, looks like all comments are currently addressed except the access token one. I couldn't find a lot of info from github via googling on this email. But is it suggesting to move the access token from the URL to the request header? Does your DB Trigger function have the following in its http.post method:
```
    headers: {
      access_token: [githubToken],
    },
```

File: source/includes/steps-tutorial-backend.yaml [hunk 1]
```diff
@@ -217,25 +239,31 @@ content: |
...
          encodeBodyAsJSON: true,
          headers: {
            access_token: [githubToken],
          },
-       });
+       })
+      .then(response => {
+        // The response body is encoded as raw BSON.Binary. Parse it to JSON.
+        const ejson_body = EJSON.parse(response.body.text());
+        return ejson_body;
```
Comment (cbush) [inline source/includes/steps-tutorial-backend.yaml:264]: Does returning this result in it being logged? Or do you have to explicitly console.log() it?

File: source/includes/steps-tutorial-backend.yaml [hunk 2]
```diff
@@ -217,25 +239,31 @@ content: |
   .. code-block:: javascript
     
      exports = function (changeEvent) { // the changeEvent refers to the document inserted
-       const githubToken = context.values.get("<githubTokenValue>"); // your Github access token
+       const githubToken = context.values.get("myGithubTokenVal"); // your Github access token
 
        const url = changeEvent.fullDocument.url; // the url of the issue
 
-       const http = context.services.get("<yourHTTPService>");
+       const http = context.services.get("redesigned-computing-machine"); // replace "redesigned-computing-machine" with the name of your HTTP Service from Step 1, if you chose a different name
```
Comment (cbush) [inline source/includes/steps-tutorial-backend.yaml:246]: Why not "github"?
```suggestion
       const http = context.services.get("github"); // replace "github" with the name of your HTTP Service from Step 1, if you chose a different name
```

File: source/includes/steps-tutorial-backend.yaml [hunk 3]
```diff
@@ -114,14 +126,16 @@ content: |
           timestamp: new Date()
         }
         // insert a document into your collection
-        const insertDoc = await context.services.get("<clusterName>").db("<dbName>").collection("<collectionName>").insertOne(issueDoc);
+        const insertDoc = await context.services.get("mongodb-atlas").db("tutorials").collection("myGithubIss").insertOne(issueDoc);
```
Comment (cbush) [inline source/includes/steps-tutorial-backend.yaml:129]: To match the other tutorial apps: (+ Nitpick: what is "Iss"?)
```suggestion
        const insertDoc = await context.services.get("mongodb-atlas").db("tracker").collection("myGithubIss").insertOne(issueDoc);
```

File: source/includes/steps-tutorial-backend.yaml [hunk 4]
```diff
@@ -265,10 +307,17 @@ content: |
   of the scheduled trigger. Finally, set the event type as "Function"
   and paste the following code into your scheduled trigger function.
 
+  .. note::
+
+     In the code block below, a message is printed to the console. To
+     further expand on your MongoDB Realm application, you can remove
```
Comment (cbush) [inline source/includes/steps-tutorial-backend.yaml:313]:
```suggestion
     further expand on your MongoDB Realm application, you could, for example, remove
```

File: source/includes/steps-tutorial-backend.yaml (hunk 1 above)
Comment (mongomoe) [reply source/includes/steps-tutorial-backend.yaml:265]: Fixed - looks like you do have to explicitly log it, so I'm logging it now

File: source/includes/steps-tutorial-backend.yaml [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+       const http = context.services.get("github"); // replace "github" with the name of your HTTP Service from Step 1, if you chose a different name
+
+       const githubURL = `${url}/comments`;
+
+       return http.post({
+         url: githubURL,
+         body: {
+           body:
+             "Hi this is the Github issue tracker bot for <github username>, thank you for leaving an issue. We will respond to your issue shortly. For now, check out  stackoverflow  for helpful tips: https://stackoverflow.com/questions/tagged/mongodb",
```
Comment (cbush) [inline source/includes/steps-tutorial-backend.yaml]:
```suggestion
             "Hi, this is the Github issue tracker bot for <github username>. Thank you for leaving an issue. We will respond to your issue shortly. For now, check out stackoverflow for helpful tips: https://stackoverflow.com/questions/tagged/mongodb",
```

File: source/includes/steps-tutorial-backend.yaml [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+title: Confirm That Your DB Trigger Function Worked
+ref: comment-db-trigger-fn
+content: |
+
+  Testing that your Database Trigger Function worked is the next step in
+  building your Task Tracker Application. Create a new issue on your
+  Github repository. When there is a new issue, your webhook inserts a
+  document into your collection, invoking the database trigger function.
+  View the inserted document by navigating to the "Atlas" tab on the top
+  of your MongoDB Realm UI. Click the "collections" button for your
```
Comment (cbush) [inline source/includes/steps-tutorial-backend.yaml:295]:
```suggestion
  of your MongoDB Realm UI. Click the :guilabel:`Collections` button for your
```

**Pull request merged**